modules/openapi-generator/src/main/resources/typescript-fetch/modelGeneric.mustache

```diff
@@ -101,17 +102,28 @@ export function {{classname}}FromJSONTyped(json: any, ignoreDiscriminator: boole
     return json;
     {{/hasVars}}
 }
-
 export function {{classname}}ToJSON(value?: {{classname}} | null): any {
+    return {{classname}}ToJSONTyped(value);
+}
+export function {{classname}}ToJSONTyped(value?: {{classname}} | null, ignoreDiscriminator = false): any {
     {{#hasVars}}
     if (value === undefined) {
         return undefined;
     }
     if (value === null) {
         return null;
     }
+{{#discriminator}}
+    if (!ignoreDiscriminator) {
+{{#discriminator.mappedModels}}
+        if (json['{{discriminator.propertyName}}'] === '{{mappingName}}') {
```

Comment:
I wonder how this works. It looks like the „json“ variable is not defined here, or am I missing something?
Did you run the generated code and confirm the correct behavior?